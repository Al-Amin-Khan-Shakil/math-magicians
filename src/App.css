.App {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100%;
}

.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100vw;
  height: 88px;
  background-color: #c8d6e5;
  padding: 0 2vw;
}

.logo {
  font-size: 2.5rem;
  font-weight: 800;
  color: #0a3d62;
}

.navbar ul {
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.navbar ul li {
  list-style-type: none;
  font-size: 1.3rem;
  font-weight: 500;
  padding: 6px 4vw;
  border-right: 2px solid #006266;
}

.navbar ul li:last-child {
  border: none;
  padding-right: 2vw;
}

.navlink {
  text-decoration: none;
  color: #0a3d62;
}

.active {
  color: #ffa502;
}

.home-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  row-gap: 3vh;
  padding: 15vh 4vw;
}

.home-title {
  color: #0a3d62;
  font-size: 2.3rem;
  margin: 3vh 0;
}

.home-paragraph {
  font-size: 1.2rem;
  color: #576574;
}
