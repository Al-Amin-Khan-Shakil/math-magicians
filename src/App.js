import './App.css';
import { Routes, Route, NavLink } from 'react-router-dom';
import CalculatorButtons from './components/CalculatorUI';
import Quotes from './components/Quotes';
import Home from './components/Home';

function App() {
  return (
    <>
      <nav className="navbar">
        <div className="logo">Math Magicians</div>
        <ul>
          <li>
            <NavLink to="/" className="navlink">Home</NavLink>
          </li>
          <li>
            <NavLink to="/calculator" className="navlink">Calculator</NavLink>
          </li>
          <li>
            <NavLink to="/quotes" className="navlink">Quotes</NavLink>
          </li>
        </ul>
      </nav>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/calculator" element={<CalculatorButtons />} />
        <Route path="/quotes" element={<Quotes />} />
      </Routes>
    </>
  );
}

export default App;
