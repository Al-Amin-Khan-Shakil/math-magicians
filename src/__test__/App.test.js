import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';

const mockApp = () => (
  <BrowserRouter>
    <App />
  </BrowserRouter>
);

it('Test App component', () => {
  const tree = renderer.create(mockApp()).toJSON();
  expect(tree).toMatchSnapshot();
});
