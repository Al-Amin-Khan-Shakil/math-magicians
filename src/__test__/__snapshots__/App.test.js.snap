// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component 1`] = `
Array [
  <nav
    className="navbar"
  >
    <div
      className="logo"
    >
      Math Magicians
    </div>
    <ul>
      <li>
        <a
          aria-current="page"
          className="navlink active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="navlink"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="navlink"
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="home-wrapper"
  >
    <h1
      className="home-title"
    >
      Experience Math Like Never Before with Math Magicians!
    </h1>
    <p
      className="home-paragraph"
    >
      Welcome to Math Magicians, where numbers come to life, equations dance at your fingertips, and math becomes an enchanting journey. Our cutting-edge calculator app is your passport to a world where math isn't just about solving problems; it's about exploring the wonders of the numerical universe.
    </p>
    <p
      className="home-paragraph"
    >
      In this digital realm, math transcends the mundane and transforms into an art form. With Math Magicians, you'll discover that math is not just a tool for daily tasks; it's a gateway to uncover the beauty and elegance hidden within each equation.
    </p>
  </div>,
]
`;
